mantidproject/mantid
# Add optional flags to clang-tidy

**Description of work.**
This PR allows a developer to select if they want to automatically fix clang-tidy issue, and which checks to carry out, via cmake. This should make it easier for any developer to run clang-tidy in the future

**To test:**
In the cmake-gui, search for "clang". Enable clang tidy and configure, there should be a checks parameter which contains default checks, and "apply_clang_tidy_fix". If you select apply and configure, the CMAKE_CXX_CLANG_TIDY should be appended with ";-fix" - this applies fixes to issues raised during clang-tidy checks.

Change "CLANG_TIDY_CHECKS" and configure. checks that the CMAKE_CXX_CLANG_TIDY updates to contain the same string.

Unchecking ENABLE_CLANG_TIDY should uncheck APPLY and set CHECKS to an empty string.

Fixes #25267 

*This does not require release notes* because **internal change to development framework**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
